src/Tmds.ExecFunction/ExecFunction.cs

```diff
@@ -282,12 +282,109 @@ private static string[] GetApplicationArguments()
                 string commandLine = Marshal.PtrToStringAuto(ptr);
                 return CommandLineToArgs(commandLine);
             }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                // The following logic is based on https://gist.github.com/nonowarn/770696
+                // Set up the mib array and the query for process maximum args size
+                var mib = new int[3];
+                int mibLength = 2;
+                mib[0] = MACOS_CTL_KERN;
+                mib[1] = MACOS_KERN_ARGMAX;
+
+                int size = IntPtr.Size / 2;
+                int argmax = 0;
+                var argv = new List<string>();
+
+                var mibHandle = GCHandle.Alloc(mib, GCHandleType.Pinned);
+                try
+                {
+                    var mibPtr = mibHandle.AddrOfPinnedObject();
+                    
+                    // Get the process args size
+                    SysCtl(mibPtr, mibLength, ref argmax, ref size, IntPtr.Zero, 0);
+
+                    // Get the PID so we can query this process' args
+                    var pid = Process.GetCurrentProcess().Id;
+
+                    // Now read the process args into the allocated space
+                    IntPtr procargs = Marshal.AllocHGlobal(argmax);
+                    try
+                    {
+                        mib[0] = MACOS_CTL_KERN;
+                        mib[1] = MACOS_KERN_PROCARGS2;
+                        mib[2] = pid;
+                        mibLength = 3;
+
+                        SysCtl(mibPtr, mibLength, procargs, ref argmax, IntPtr.Zero, 0);
+
+                        // The memory block we're reading is a series of null-terminated strings
+                        // that looks something like this:
+                        //
+                        // | argc      | <int>
+                        // | exec_path | ... \0
+                        // | argv[0]   | ... \0
+                        // | argv[1]   | ... \0
+                        //   ...
+
+                        // Read argc
+                        var argc = Marshal.ReadInt32(procargs);
+
+                        // Skip over argc
+                        var argvPtr = IntPtr.Add(procargs, sizeof(int));
+
+                        // Skip over exec_path
+                        var offset = 0;
+                        while (Marshal.ReadByte(argvPtr, offset) != 0) { offset++; }
+                        while (Marshal.ReadByte(argvPtr, offset) == 0) { offset++; }
+                        argvPtr = IntPtr.Add(argvPtr, offset);
+
+                        for (var i = 0; i < argc; i++)
+                        {
+                            var arg = Marshal.PtrToStringAnsi(argvPtr);
```

Comment:
Can you find `\0` and then use the overload of `PtrToStringAnsi` that accepts a `length`?
I would like to avoid any issues where `PtrToStringAnsi` has consumed a different amount of bytes than returned by `ASCII.GetByteCount`.

Reply:
Makes sense, I updated this  👍